OpenKeychain/src/main/java/org/sufficientlysecure/keychain/pgp/PgpSignEncryptOperation.java

```diff
@@ -229,6 +226,15 @@ private PgpSignEncryptResult executeInternal(PgpSignEncryptInputParcel input, Cr
                         mProviderHelper.getCanonicalizedSecretKeyRing(signingMasterKeyId);
                 signingKey = signingKeyRing.getSecretKey(data.getSignatureSubKeyId());
 
+                if (input.getAllowedKeyIds() != null) {
+                    if (!input.getAllowedKeyIds().contains(signingMasterKeyId)) {
+                        // this key is in our db, but NOT allowed!
+                        log.add(LogType.MSG_DC_ASKIP_NOT_ALLOWED, indent + 1);
+                        log.add(LogType.MSG_DC_ERROR_NO_KEY, indent + 1);
```

Comment:
Please introduce new LogTypes here. The current ones you used are for decryption (_DC_)